droidlet/tools/hitl/nsp_retrain_infra.py

```diff
@@ -161,28 +161,39 @@ def create_mask(self, opts, batch_id, data_filepath):
-
-            # Lengthen valid and test masks to be the length of all data
-            old_mask_filepath = os.path.join(opts.sweep_config_folder, 'split_masks.pth')
-            old_masks = torch.load(old_mask_filepath)['annotated']
-            valid_mask = [x for x in old_masks['valid']]
-            valid_mask.extend([False] * (total_rows - len(valid_mask)))
-            test_mask = [x for x in old_masks['test']]
-            test_mask.extend([False] * (total_rows - len(test_mask)))
-            train_mask = [False if valid_mask[i] or test_mask[i] else True for i in range(total_rows)]  # Rest of the data are train
-        else:
-            logging.info(f"Model train/valid/test data will be resampled from combined old+new dataset in 80/10/10 split")
-
-            # Generate random splits and populate masks
-            mask_designation = ['test'] * int(total_rows * 0.1)
-            mask_designation.extend(['valid'] * int(total_rows * 0.1))
-            mask_designation.extend(['train'] * (total_rows - len(mask_designation)))
-            random.shuffle(mask_designation)
-            train_mask = [True if x == 'train' else False for x in mask_designation]
-            valid_mask = [True if x == 'valid' else False for x in mask_designation]
-            test_mask = [True if x == 'test' else False for x in mask_designation]
+
+        # Old masks are static, just extend to be the length of the new dataset
+        old_mask_filepath = os.path.join(opts.sweep_config_folder, 'split_masks.pth')
+        old_masks = torch.load(old_mask_filepath)['annotated']
+        old_train_mask = [x for x in old_masks['train']]
+        old_train_mask.extend([False] * (total_rows - len(old_train_mask)))
+        old_valid_mask = [x for x in old_masks['valid']]
+        old_valid_mask.extend([False] * (total_rows - len(old_valid_mask)))
+        old_test_mask = [x for x in old_masks['test']]
+        old_test_mask.extend([False] * (total_rows - len(old_test_mask)))
+
+        # Set random seed for reproducability and generate masks for new data
+        random.seed(batch_id)
+        new_mask_designation = ['test'] * int(new_data_rows * 0.1)
+        new_mask_designation.extend(['valid'] * int(new_data_rows * 0.1))
+        new_mask_designation.extend(['train'] * (new_data_rows - len(new_mask_designation)))
+        random.shuffle(new_mask_designation)
+        new_masks = [False] * (total_rows - new_data_rows)  # Prepend False for old data
+        new_masks.extend(new_mask_designation)
+        new_train_mask = [True if x == 'train' else False for x in new_masks]
+        new_valid_mask = [True if x == 'valid' else False for x in new_masks]
+        new_test_mask = [True if x == 'test' else False for x in new_masks]
+
+        # Mix and match old and new data based on user input
+        if opts.retrain_data_splits[0] == 0: train_mask = old_train_mask
```

Comment:
same as above, we should try to nuke as much redundancy in code as possible.

Reply:
also addressed in the latest commit